### PR TITLE
TNO-1553,1552,1516:Create/manage folders, filter by sentiment

### PR DESCRIPTION
**Pull request by jdtoombs**
_**Known bug:**_ content not being returned by api on attached folders, Will create ticket for next sprint

Also a note for @Fosol. we are probably moving advanced search out of the sidebar so things aren't as squished.

**In this PR:**

1) can now filter out content based on their tone in the advanced search
sentiment

2) my folders section allows you to manage/update folders
foldermanage

3) add to folder button when viewing content now allows you to add content to existing folder, or a new folder
addtofolder

### CODE REVIEW

File: app/subscriber/src/features/home/utils/makeFilter.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export const makeFilter = (
     productIds: filter.productIds ?? undefined,
     publishedEndOn: filter.endDate ? filter.endDate : undefined,
     publishedStartOn: filter.startDate ? filter.startDate : undefined,
+    sentiment: filter.sentiment ?? undefined,
```
Comment (Fosol) [inline app/subscriber/src/features/home/utils/makeFilter.ts:26]: Isn't filter an array or undefined?  This doesn't seem like a relevant operation.  Just use `filter.sentiment` I think.

File: app/subscriber/src/features/my-folders/styled/MyFolders.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    overflow: hidden;
+    .re-name {
+      height: 1.5em;
+      padding: 0;
+      &:focus {
+        box-shadow: none;
+      }
+    }
+    .group {
+      background-color: #f9f9f9 !important;
```
Comment (Fosol) [inline app/subscriber/src/features/my-folders/styled/MyFolders.tsx:37]: Can you check if this is really the only way to do this?

File: libs/net/dal/Models/ContentFilter.cs [hunk 3]
```diff
@@ -110,6 +110,12 @@ public class ContentFilter : PageFilter
     /// </summary>
     public string? Section { get; set; }
 
+    /// <summary>
+    /// get/set - The content sentitment.
+    /// </summary>
+    public long[] Sentiment { get; set; }
```
Comment (Fosol) [inline libs/net/dal/Models/ContentFilter.cs:116]: Fix nullable issues.

**Review by Fosol — CHANGES_REQUESTED** [review]

File: app/subscriber/src/features/my-folders/styled/MyFolders.tsx (hunk 2 above)
Comment (jdtoombs) [reply app/subscriber/src/features/my-folders/styled/MyFolders.tsx:37]: On it

**Comment by jdtoombs** [comment]
Updated :) @Fosol

File: libs/net/dal/Models/ContentFilter.cs [hunk 4]
```diff
@@ -173,6 +173,11 @@ public class ContentFilter : PageFilter
     /// </summary>
     public string[] Sort { get; set; } = Array.Empty<string>();
 
+    /// <summary>
+    /// get/set - The content sentitment.
+    /// </summary>
+    public long[] Sentiment { get; set; } = Array.Empty<long>();
```
Comment (Fosol) [inline libs/net/dal/Models/ContentFilter.cs:179]: Is there any reason to use a `long`?  I think the value will never be more than -5 to +5.

Comment (jdtoombs) [reply libs/net/dal/Models/ContentFilter.cs]: Copy and paste d'oh mistake, int now :)

**Review by Fosol — APPROVED** [review]

**Pull request merged**